README.md

```diff
@@ -46,21 +46,43 @@ Relative import statements are resolved against with the path of the _declaring_
 
 ## Syntax details
 
+(TODO: Add syntax for module functions.)
+
 ```
-PrimaryExpression :  InlineModuleExpression
+PrimaryExpression :  ModuleBlockExpression
+                     ModuleFunctionExpression
+
+ModuleBlockExpression : `module` [no LineTerminator here] `{` ModuleBody? `}`
 
-InlineModuleExpression : module [no LineTerminator here] { Module }
+ModuleFunctionExpression : `module` [no LineTerminator here] FunctionExpression
+// ... same for AsyncFunctionExpression et al.
 ```
 
-As `module` is not a keyword in JavaScript, no newline is permitted between `module` and `{`.
+As `module` is not a keyword in JavaScript, no newline is permitted after `module`.
 
-## HTML Integration
+### Module function syntactic sugar
 
-There are 4 main integration points in the HTML spec for Module Blocks:
+Most native platforms offer parallelism with functions as the fundamental primitive, not modules. JavaScript can’t (and shouldn’t) adopt other ecosystem’s primitives directly, as there are fundamental difference. However, to allow the implementation of syntactically lightweight abstractions that aim for similar ergonomics, it makes sense to also offer module _functions_. A module function is equivalent to a module block with a single, default export:
 
-### Worker constructor 
+```js
+const m = module function(a, b, c) {
+  // ...
+}
 
-`new Worker()` currently only accepts a path to a worker file. The proposal aims to also let it accept a Module Block directly (for `{type: "module"}` workers). 
+// desugars to
+
+const m = module {
```

Comment:
I think there is no need to add a sugar for this, the desugar form looks fine too.